src/main/scala/cognite/spark/v1/TimeSeriesRelation.scala

```diff
@@ -77,7 +77,8 @@ class TimeSeriesRelation(config: RelationConfig, useLegacyName: Boolean)(val sql
     upsertAfterConflict[TimeSeries, TimeSeriesUpdate, TimeSeriesCreate, TimeSeriesResource[IO]](
       existingExternalIds,
```

Comment:
It looks to me like this could have the same implementation as Assets and Events here, where the `externalIds` are handled in `upsertAfterConflict`, but I may be wrong. If it can, we can split `upsertAfterConflict` in two and avoid the recursion.